python/example_code/dynamodb/GettingStarted/scenario_getting_started_movies.py

```diff
@@ -102,6 +102,27 @@ def create_table(self, table_name):
             return self.table
     # snippet-end:[python.example_code.dynamodb.CreateTable]
 
+    # snippet-start:[python.example_code.dynamodb.ListTables]
+    def list_tables(self):
+        """
+        Lists the Amazon DynamoDB tables for the current account.
+
+        :return: The list of tables.
+        """
+        try:
+            tables = []
+            for table in self.dyn_resource.tables.all():
+                print(table.name)
+                tables.append(table)
+        except ClientError as err:
+            logger.error(
+                "Couldn't list tables. Here's why: %s: %s",
+                err.response['Error']['Code'], err.response['Error']['Message'])
+            raise
+        else:
```

Comment:
One thing to think about here, is your code is functionally the same with or without this else (as long as you fix the indentation of the next line). This comes down mostly to personal preference and style, but some people really don't like a single line in an else at the end of a function. Not sure if there's a best practice for Python here, but ultimately it's not a big deal. Just pointing it out for you to ponder.

Reply:
Interesting point! This is pretty common in Python, so I think it's a language thing.
One subtlety that doesn't come into play in this specific example is that the else block is in the same scope as the try, so in the else block you can reference variables that are declared and initialized in the try block, whereas if you put the return outside, you'll have to also move the initialization above the try.